app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -75,35 +82,64 @@ private void applyNumberOfGroups() {
         numGroups.clearFocus();
     }
 
+    @OnCheckedChanged(R.id.autocomplete)
+    void onAutocompleteSelected(boolean checked) {
+        if (checked) {
+            if (isInputValid()) {
+                namesPerGroup.setText(getMatchingNumber(numberOfGroupsInput));
+            }
+        }
+    }
+
     @OnTextChanged(value = R.id.num_of_names_per_group, callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED)
     void afterNamesPerGroupChanged() {
-        String namesPerGroupInput = namesPerGroup.getText().toString().trim();
-        boolean isInputValid = isInputValid(namesPerGroupInput);
-        if (namesPerGroup.isFocused() && isInputValid) {
+        namesPerGroupInput = namesPerGroup.getText().toString().trim();
+        boolean isInputValid = isInputValid();
+        if (namesPerGroup.isFocused() && isInputValid && autocomplete.isChecked()) {
             numGroups.setText(getMatchingNumber(namesPerGroupInput));
         }
         dialog.getActionButton(DialogAction.POSITIVE).setEnabled(isInputValid);
+        maybeShowUnevenGroupsWarning();
     }
 
     @OnTextChanged(value = R.id.num_of_groups, callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED)
     void afterGroupNumChanged() {
-        String numberOfGroupsInput = numGroups.getText().toString().trim();
-        boolean isInputValid = isInputValid(numberOfGroupsInput);
-        if (numGroups.isFocused() && isInputValid) {
+        numberOfGroupsInput = numGroups.getText().toString().trim();
+        boolean isInputValid = isInputValid();
```

Comment:
Same feedback as above. The logic should be as follows:
1. Parse out the input
2. See if the autocomplete is turned on and act accordingly
3. Turn the action button on/off